src/pages/LogInWithShortLivedAuthTokenPage.tsx

```diff
@@ -18,12 +18,18 @@ type LogInWithShortLivedAuthTokenPageProps = PlatformStackScreenProps<PublicScre
 
 function LogInWithShortLivedAuthTokenPage({route}: LogInWithShortLivedAuthTokenPageProps) {
     const {shortLivedAuthToken = '', shortLivedToken = '', authTokenType, exitTo, error} = route?.params ?? {};
-    const [account] = useOnyx(ONYXKEYS.ACCOUNT);
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT, {canBeMissing: true});
+    const [lastShortLivedToken] = useOnyx(ONYXKEYS.NVP_SHORT_LIVED_TOKEN, {canBeMissing: true});
 
     useEffect(() => {
         // We have to check for both shortLivedAuthToken and shortLivedToken, as the old mobile app uses shortLivedToken, and is not being actively updated.
         const token = shortLivedAuthToken || shortLivedToken;
 
+        // This is to prevent re-authenticating when logging out if the initial URL (deep link) hasn't changed.
+        if (token && token === lastShortLivedToken) {
```

Comment:
```suggestion
        if (token && token === lastShortLivedToken) {
```

Do we really need the `token &&` part?

Reply:
Went through all automated and manual tests and we don't actually need it. Thanks for catching that! :)